yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -397,15 +395,26 @@ func (p *ParserIssueDetector) genericIssues(query string) ([]QueryIssue, error)
 		return result, fmt.Errorf("error traversing parse tree message: %w", err)
 	}
 
-	for _, unsupportedConstruct := range unsupportedConstructs {
-		switch unsupportedConstruct {
-		case ADVISORY_LOCKS_NAME:
-			result = append(result, NewAdvisoryLocksIssue(DML_QUERY_OBJECT_TYPE, "", query))
-		case SYSTEM_COLUMNS_NAME:
-			result = append(result, NewSystemColumnsIssue(DML_QUERY_OBJECT_TYPE, "", query))
-		case XML_FUNCTIONS_NAME:
-			result = append(result, NewXmlFunctionsIssue(DML_QUERY_OBJECT_TYPE, "", query))
+	xmlIssueAdded := false
+	for _, detector := range detectors {
+		issues := detector.GetIssues()
+		for _, issue := range issues {
+			if issue.Type == XML_FUNCTIONS {
+				if xmlIssueAdded {
+					// currently, both FuncCallDetector and XmlExprDetector can detect XMLFunctionsIssue
+					// but we want to only return one XMLFunctionsIssue.
+					// TODO: refactor to avoid this
+					// Possible Solutions:
+					// 1. Have a dedicated detector for XMLFunctions and Expressions so that a single issue is returned
+					// 2. Separate issue types for XML Functions and XML expressions.
+					continue
+				} else {
+					xmlIssueAdded = true
+				}
```

Comment:
maybe instead you can directly use the `lo.UniqBy` on the `result` based on `issue.Type`

Reply:
Don't want to blindly do this. This is just a workaround for xmlfunctions right now. 
If there are multiple issues of the same type, we can't blindly pick 1, we will have to essentially merge the issues (along with the details field).
 
If we encounter more such cases(other than XML) in the future, we need to figure out one of those solutions listed above. Did not want to do that work now as part of this refactor.

Reply:
> If there are multiple issues of the same type, we can't blindly pick 1, we will have to essentially merge the issues (along with the details field).

Hmm makes sense when we start adding the function names or actual constructs in detail from detector layer.